### PR TITLE
istioctl: add a flag to delete all waypoints in a namespace

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**
Fix https://github.com/istio/istio/pull/47539#issuecomment-1778546940

### CODE REVIEW

**Comment by hanxiaop** [comment]
I'll add this to the integration test once https://github.com/istio/istio/pull/47615 is merged.

File: istioctl/pkg/waypoint/waypoint.go [hunk 1]
```diff
@@ -158,12 +162,38 @@ func Cmd(ctx cli.Context) *cobra.Command {
...
 
   # Delete a waypoint by name, which can obtain from istioctl x waypoint list 
-  istioctl x waypoint delete waypoint-name --namespace default`,
+  istioctl x waypoint delete waypoint-name --namespace default
+
+  # Delete several waypoints by name
+  istioctl x waypoint delete waypoint-name1 waypoint-name2 --namespace default
+
+  # Delete all waypoints in a specific namespace
+  istioctl x waypoint delete --all --namespace default`,
```
Comment (GregHanson) [inline istioctl/pkg/waypoint/waypoint.go:171]: what's the behavior if user does `istioctl x waypoint delete --all -A`? I imagine that is not a scenario we do not want to support?

Comment (hanxiaop) [reply istioctl/pkg/waypoint/waypoint.go:171]: Yes, it's not supported in `delete` command, it's only supported in `list` command.

Comment (hanxiaop) [reply istioctl/pkg/waypoint/waypoint.go:171]:
```
istioctl x waypoint delete  -n bookinfo --all -A
Error: unknown shorthand flag: 'A' in -A
```

Comment (GregHanson) [reply istioctl/pkg/waypoint/waypoint.go:171]: sounds good

**Review by GregHanson — APPROVED** [review]

**Comment by hanxiaop** [comment]
Interesting, where is the `integ-telemetry-discovery_istio` test going?

**Comment by hanxiaop** [comment]
@istio/wg-test-and-release-maintainers PTAL, thanks

**Review by jwendell — APPROVED** [review]

**Pull request merged**